packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts

```diff
@@ -298,6 +252,160 @@ function buildInterrupt({
   };
 }
 
+async function getRespWithDocsToDownload({
+  cxId,
+  patientId,
+  requestId,
+  response,
+}: OutboundDocQueryRespParam): Promise<OutboundDocumentQueryResp[]> {
+  const respWithDocsToDownload: OutboundDocumentQueryResp[] = [];
+
+  await executeAsynchronously(
+    response,
+    async gwResp => {
+      const resultsWithMetriportId = await updateDocumentReferencesWithMetriportId({
+        cxId,
+        patientId,
+        requestId,
+        response: gwResp,
+      });
+      const docRefs = resultsWithMetriportId.flatMap(result => result.documentReference ?? []);
```

Comment:
nit: also no need for the `[]` fallback

Reply:
ditto documentReference can be undefined